qlib/log.py

```diff
@@ -17,6 +17,7 @@ def __new__(cls, name, bases, dict):
         wrapper_dict = logging.Logger.__dict__.copy()
         wrapper_dict.update(dict)
         wrapper_dict["__doc__"] = logging.Logger.__doc__
+        del wrapper_dict["__reduce__"]  # make Logger object can be pickled
```

Comment:
I'm little confused about the current implementation.

If the problem is that copying the pickle related magic methods will result in error when pickling, why don't we just exclude them when copying methods dict.

![image](https://user-images.githubusercontent.com/465606/116550644-97c5ad80-a929-11eb-92ff-87cfcd2723f2.png)